### PR TITLE
Added: Chevron to Diagnostic Items & Drop Shadow

### LINKED ISSUE #1796: Health check diagnostics list - UI improvement
UI improvement to make it more obvious that the items in the list can be clicked to open the diagnostic page.

## Design

Arrow added right hand side and drop-shadow-small on all items. Hover state behavior same as current, icon colour should change to neutral strong on hover, same as the text.

[Figma link](https://www.figma.com/design/hsUiqro210T3tsK1F9l1fQ/%F0%9F%93%B1%F0%9F%A9%BAHealth-Check?node-id=1-2954&t=OTu6Knvb49PbbNxF-1)

Image

### PR DESCRIPTION
**Pull request by Sewer56**
fixes #1796

- https://github.com/Nexus-Mods/NexusMods.App/issues/1796

image

Getting that shadow to appear was quite the hassle (and wasted a lot of my time!!) due to how sizing works within the panel.

Notably it also it did not want to apply to the button [NavigationControl] itself; so I reserved some margin area for the shadow. It does mean the inner content is smaller, but the difference is not very big.

### CODE REVIEW

File: src/NexusMods.App.UI/Controls/Diagnostics/DiagnosticEntryView.axaml [hunk 1]
```diff
@@ -6,23 +6,32 @@
...
+                        VerticalAlignment="Center">
+
+                <icons:UnifiedIcon x:Name="SeverityIcon" Size="20" />
+                <TextBlock x:Name="DescriptionText"
+                           Theme="{StaticResource BodyMDBoldTheme}"
+                           VerticalAlignment="Center"/>
+            
+            </StackPanel>
+            
+            <chevron:Chevron Grid.Column="1"
```
Comment (Al12rs) [inline src/NexusMods.App.UI/Controls/Diagnostics/DiagnosticEntryView.axaml:30]: This should just be a `UnifiedIcon`, it does not need to be a dedicated control. If you need pointerover put it in a border or button.

File: src/NexusMods.App.UI/Controls/Chevron/Chevron.axaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<UserControl xmlns="https://github.com/avaloniaui"
```
Comment (Al12rs) [inline src/NexusMods.App.UI/Controls/Chevron/Chevron.axaml:1]: This doesn't need to exist

**Review by Al12rs — CHANGES_REQUESTED** [review]

File: src/NexusMods.App.UI/NexusMods.App.UI.csproj [hunk 3]
```diff
@@ -641,6 +641,7 @@
       <UpToDateCheckInput Remove="Controls\Buttons\StandardButtonTheme.axaml" />
       <UpToDateCheckInput Remove="RightContent\Home\HomeView.axaml" />
       <UpToDateCheckInput Remove="Controls\Badge\Badge.axaml" />
+      <UpToDateCheckInput Remove="Controls\Chevron\Chevron.axaml" />
```
Comment (Al12rs) [inline src/NexusMods.App.UI/NexusMods.App.UI.csproj:644]: Leftover from old files, can be removed

**Comment by insomnious** [comment]
image

File: src/NexusMods.App.UI/Controls/Diagnostics/DiagnosticEntryView.axaml [hunk 4]
```diff
@@ -6,23 +6,30 @@
...
+                <panels:FlexPanel x:Name="EntryFlexPanel" Direction="Row" ColumnSpacing="8">
+                    <icons:UnifiedIcon x:Name="SeverityIcon" Size="20" HorizontalAlignment="Left" />
+                    <TextBlock x:Name="DescriptionText"
+                               Theme="{StaticResource BodyMDNormalTheme}"
+                               VerticalAlignment="Center"
+                               TextTrimming="CharacterEllipsis"
+                               TextWrapping="NoWrap"
+                               />
+                    <icons:UnifiedIcon Classes="ChevronRight" Size="32" HorizontalAlignment="Right" VerticalAlignment="Center"/>
+                </panels:FlexPanel>
```
Comment (Al12rs) [inline src/NexusMods.App.UI/Controls/Diagnostics/DiagnosticEntryView.axaml:29]: You shouldn't be using Horizontal and vertical alignment on the child elements when using `FlexPanel` as it basically overrides the FlexPanel own layout.

To get the distribution you want you should try playing with `JustifyContent` property on the FlexPanel. It determines the distribution of items along the main axis (horizontal in this case).

I think SpaceBetween is likely what we want with Flex.Grow=1 on the TextBlock to allow it to take up as much space in the middle as available.

**Review by Al12rs — CHANGES_REQUESTED** [review]
Some adjustments for FlexPanel. I think in this case a Grid with `Auto,*,Auto` columns and manual padding between them could also work.

**Comment by insomnious** [comment]
Removed the extra alignments and moved all styles to the Style file. Laurence noted that the MaxWidth wasn't set anywhere and so it doesn't [stay in middle of view](https://www.figma.com/design/hsUiqro210T3tsK1F9l1fQ/%F0%9F%93%B1%F0%9F%A9%BAHealth-Check?node-id=204-3029&t=AlUpeSqXdI3i5Vjb-1). He's happy with how it looks now but I'm not happy it's hard coded into the `DiagnosticListView.axaml` file along with some other styles that are already there. It'll serve it's purpose for now until we move to phase 2 of Health Check (as per [Figma](https://www.figma.com/design/hsUiqro210T3tsK1F9l1fQ/%F0%9F%93%B1%F0%9F%A9%BAHealth-Check?node-id=1-3041&t=AlUpeSqXdI3i5Vjb-1)), and I've left a `TODO` in the code.

**Comment by Al12rs** [comment]
@insomnious Could you update this ticket with the stuff that was covered in this PR?
https://github.com/Nexus-Mods/NexusMods.App/issues/1091

**Review by Al12rs — APPROVED** [review]
Code looks good, did lausandy already look at the UI results?

**Review by captainsandypants — APPROVED** [review]
Looks great!

**Pull request merged**